File: src/main/java/com/github/iamniklas/liocore/led/json/interpreter/strategies/IInterpreterStrategy.java
package com.github.iamniklas.liocore.led.json.interpreter.strategies;

import com.github.iamniklas.colorspaces.ColorRGB;
import com.github.iamniklas.liocore.led.json.LEDJsonProcedure;

public interface IInterpreterStrategy {
    ColorRGB[] interpretLine(String _line);
    LEDJsonProcedure interpretJson(String _json);
}

File: src/main/java/com/github/iamniklas/liocore/led/json/interpreter/strategies/IInterpreterStrategy.kt
package com.github.iamniklas.liocore.led.json.interpreter.strategies

import com.github.iamniklas.liocore.led.ColorRGB
import com.github.iamniklas.liocore.led.json.LEDJsonProcedure

interface IInterpreterStrategy {
    fun interpretLine(_line: String?): Array<ColorRGB>
    fun interpretJson(_json: String?): LEDJsonProcedure?
}